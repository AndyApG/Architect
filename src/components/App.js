import React from 'react';
import { motion } from 'framer-motion';
import { compose } from 'recompose';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { withRouter } from 'react-router-dom';
import { isMacOS } from '../utils/platform';
import { AppErrorBoundary } from './Errors';
import DialogManager from './DialogManager';
import Home from './Home/Home';

const appVariants = {
  visible: { opacity: 1, duration: 2 },
  hidden: { opacity: 0, duration: 2 },
};

const App = (props) => {
  const appClasses = cx(
    'app',
    {
      'app--macos': isMacOS(),
    },
  );

  // we can use location and history for router

  return (
    <React.Fragment>
      {isMacOS() &&
        <div className="electron-titlebar" />
      }
      <motion.div
        className={appClasses}
        variants={appVariants}
        initial="hidden"
        animate="visible"
      >
        <AppErrorBoundary>
          <Home />
        </AppErrorBoundary>
      </motion.div>
      <DialogManager />
    </React.Fragment>
  );
};

App.propTypes = {
  // children: PropTypes.element,
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
};

App.defaultProps = {
  children: null,
};

export { App };

export default compose(
  withRouter,
)(App);
