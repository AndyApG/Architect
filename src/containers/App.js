import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { compose } from 'recompose';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import cx from 'classnames';
import { actionCreators as protocolActions } from '../ducks/modules/protocol';

require('../styles/main.scss');

const isAtIndex = pathname => pathname === '/';

const App = ({ children, location: { pathname }, resetActiveProtocol }) => {
  const appClasses = cx(
    'app',
    {
      'app--start': isAtIndex(pathname),
    },
  );

  return (
    <div className={appClasses}>
      <div className="app__logo" onClick={resetActiveProtocol} />
      { children }
    </div>

  );
};

App.propTypes = {
  children: PropTypes.any,
  location: PropTypes.object.isRequired,
  resetActiveProtocol: PropTypes.func.isRequired,
};

App.defaultProps = {
  children: null,
};

const mapDispatchToProps = dispatch => ({
  resetActiveProtocol: bindActionCreators(protocolActions.resetProtocol, dispatch),
});

export { App };

export default compose(
  withRouter,
  connect(null, mapDispatchToProps),
)(App);
