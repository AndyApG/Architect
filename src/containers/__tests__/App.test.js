/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../App';

const mockProps = {
  location: { pathname: '' },
};

describe('<App />', () => {
  it('can render', () => {
    const component = shallow(<App {...mockProps} />);

    expect(component).toMatchSnapshot();
  });

  it('has `.app--start` class if path is root', () => {
    const componentAtFoo = shallow(<App location={{ pathname: '/foo' }} />);
    expect(componentAtFoo.hasClass('app--start')).toBe(false);

    const componentAtRoot = shallow(<App location={{ pathname: '/' }} />);
    expect(componentAtRoot.hasClass('app--start')).toBe(true);
  });
});
