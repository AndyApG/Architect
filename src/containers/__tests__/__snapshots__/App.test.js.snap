// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    location={
        Object {
            "pathname": "",
          }
    }
    resetActiveProtocol={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="app__title-bar"
          onClick={[Function]}
>
          Architect app
</div>,
        null,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Architect app",
          "className": "app__title-bar",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Architect app",
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="app__title-bar"
            onClick={[Function]}
>
            Architect app
</div>,
          null,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Architect app",
            "className": "app__title-bar",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Architect app",
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
